Rewritten from scratch

Abbe98 opened the pull request:
In addition to visual change:

 - Tries to check if the image is already imported.
 - Picks up location label from Mapzen.

It uses the same URL parameter and should be fully backwards capable.
![image](https://cloud.githubusercontent.com/assets/2631719/26819650/1d9d7d8a-4aa0-11e7-95b8-43c95820bed5.gif)

Abbe98 committed:
rewritten from scratch

lokal-profil commented on simpleUpload/index.html:
```diff
@@ -1,96 +1,35 @@
-    <title>Mapillary2Commons</title>
+	<head>
+		<meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta charset="utf-8" />
```
Strange mixture of indents

Abbe98 replied:
Noted

Abbe98 committed:
replace ES6 arrows to support Edge/IE

Abbe98 committed:
Fix edge-case where Mapzen does not find a location

Abbe98 committed:
minor workaround for IE/Edge CSS bug

Abbe98 commented:
Tested in IE/Edge, Firefox, and Opera/Chrome.

lokal-profil commented on simpleUpload/index.html:
```diff
@@ -1,96 +1,35 @@
-                Ever found a beautiful image on <a href="https://www.mapillary.com">Mapillary</a>
-                and thought <i>That would look great on <a href="https://commons.wikimedia.org">Wikimedia Commons</a></i>?
-            </p>
-            <p> Simply enter the image key or url below and click <i>Check!</i>.
```
Any reason why you completely dropped this information bit?

Abbe98 replied:
No, not at all (in both cases), I started from scratch so I just didn't  think about it.

lokal-profil commented on simpleUpload/index.html:
```diff
@@ -1,96 +1,35 @@
-                            you'll be prompted to enter one.
-                        </li>
-                        <li>
-                            You can also pass this page a url directly using
```
Also this info

lokal-profil reviewed (commented):
Sorry for taking so long to get back on this.

Thanks for giving this a work-over and making it work after the API deprecation.

A few inline questions. Also what happens if mapzen fails to find a location (i.e. are you stil prevented from going on without a location)?

lokal-profil commented:
Temporarily up at https://tools.wmflabs.org/mapillary-commons/ver2/ for testing

Abbe98 commented:
Adding the missing information and cleaning up the weird indents once I'm done for the day :-)

If Mapzen isn't able to (re)geocode the location, the user must enter a location description before the upload button gets enabled. Example: 
https://tools.wmflabs.org/mapillary-commons/ver2/?mapillary=https://www.mapillary.com/map/im/e814RJRxf8cnqn4OEFFMnQ

lokal-profil commented:
@Abbe98

In my mobile at least if I click the Upload button it reloads the page.  Could it display a message asking for location input instead?

Also noticed that if I fill in a location and then erase it it still uses the first letters of the deleted string afterwards.

Abbe98 committed:
improve manual location description fallback

Abbe98 committed:
move and add missing information

Abbe98 committed:
minor HTML linting

Abbe98 commented:
Manual location descriptions should work now and if the description is to short or does not exists a error message will replace the file name. (The Upload button does no longer reload the page.) 

The existing information about OAuth was moved to a fixed box, earlier you could miss it if the `mapillary` parameter was in use.

In the place of OAuth information there is now the old `lead` text.

I added a link to source code and API docs at the bottom of the header and changed the README to include information about the `mapillary` parameter.

lokal-profil commented:
Thank you!

lokal-profil commented:
Deployed

Abbe98 commented:
Amazing, thanks taking the time!
